```
Error running mint automation
+ mkdir -p 10016-d39a79e/gopath/src/github.com/minio
+ git -C 10016-d39a79e/gopath/src/github.com/minio clone --quiet https://github.com/minio/minio.git
+ git -C 10016-d39a79e/gopath/src/github.com/minio/minio remote add minio https://github.com/minio/minio.git
+ git -C 10016-d39a79e/gopath/src/github.com/minio/minio fetch --quiet minio pull/10016/head:pr10016
+ git -C 10016-d39a79e/gopath/src/github.com/minio/minio checkout --quiet pr10016
+ GO111MODULE=on
+ GOPATH=/home/mint-auto/mint-auto/10016-d39a79e/gopath
+ make -C 10016-d39a79e/gopath/src/github.com/minio/minio --quiet
go: downloading github.com/json-iterator/go v1.1.10
go: downloading github.com/minio/minio-go/v6 v6.0.58-0.20200612001654-a57fec8037ec
go: downloading google.golang.org/api v0.5.0
go: downloading cloud.google.com/go v0.39.0
go: downloading github.com/minio/cli v1.22.0
go: downloading github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: downloading github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: downloading github.com/gorilla/rpc v1.2.0
go: extracting github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: extracting github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: downloading github.com/miekg/dns v1.1.8
go: downloading github.com/gorilla/mux v1.7.4
go: extracting github.com/json-iterator/go v1.1.10
go: downloading go.etcd.io/etcd/v3 v3.3.0-rc.0.0.20200707003333-58bb8ae09f8e
go: extracting github.com/gorilla/mux v1.7.4
go: downloading github.com/mitchellh/go-homedir v1.1.0
go: extracting github.com/mitchellh/go-homedir v1.1.0
go: extracting cloud.google.com/go v0.39.0
go: downloading github.com/klauspost/compress v1.10.3
go: extracting github.com/miekg/dns v1.1.8
go: extracting github.com/minio/cli v1.22.0
go: downloading github.com/willf/bloom v2.0.3+incompatible
go: downloading github.com/shirou/gopsutil v2.20.3-0.20200314133625-53cec6b37e6a+incompatible
go: extracting github.com/gorilla/rpc v1.2.0
go: extracting github.com/willf/bloom v2.0.3+incompatible
go: extracting github.com/minio/minio-go/v6 v6.0.58-0.20200612001654-a57fec8037ec
go: downloading github.com/minio/highwayhash v1.0.0
go: downloading github.com/colinmarc/hdfs/v2 v2.1.1
go: downloading golang.org/x/crypto v0.0.0-20200622213623-75b288015ac9
go: downloading github.com/google/uuid v1.1.1
go: extracting github.com/shirou/gopsutil v2.20.3-0.20200314133625-53cec6b37e6a+incompatible
go: extracting go.etcd.io/etcd/v3 v3.3.0-rc.0.0.20200707003333-58bb8ae09f8e
go: downloading github.com/dchest/siphash v1.2.1
go: downloading google.golang.org/grpc v1.26.0
go: downloading github.com/secure-io/sio-go v0.3.0
go: extracting github.com/colinmarc/hdfs/v2 v2.1.1
go: extracting github.com/google/uuid v1.1.1
go: downloading github.com/cheggaaa/pb v1.0.28
go: extracting github.com/dchest/siphash v1.2.1
go: extracting github.com/minio/highwayhash v1.0.0
go: downloading github.com/montanaflynn/stats v0.5.0
go: extracting github.com/secure-io/sio-go v0.3.0
go: downloading github.com/Azure/azure-pipeline-go v0.2.1
go: extracting github.com/cheggaaa/pb v1.0.28
go: downloading gopkg.in/olivere/elastic.v5 v5.0.80
go: extracting github.com/montanaflynn/stats v0.5.0
go: extracting github.com/Azure/azure-pipeline-go v0.2.1
go: downloading github.com/elazarl/go-bindata-assetfs v1.0.0
go: extracting github.com/elazarl/go-bindata-assetfs v1.0.0
go: downloading golang.org/x/sys v0.0.0-20200610111108-226ff32320da
go: extracting golang.org/x/crypto v0.0.0-20200622213623-75b288015ac9
go: extracting gopkg.in/olivere/elastic.v5 v5.0.80
go: extracting google.golang.org/grpc v1.26.0
go: downloading gopkg.in/jcmturner/gokrb5.v7 v7.3.0
go: downloading github.com/gorilla/handlers v1.4.2
go: downloading github.com/djherbis/atime v1.0.0
go: downloading golang.org/x/net v0.0.0-20200602114024-627f9648deb9
go: downloading github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: extracting github.com/djherbis/atime v1.0.0
go: extracting github.com/gorilla/handlers v1.4.2
go: extracting github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: downloading github.com/go-sql-driver/mysql v1.5.0
go: downloading github.com/hashicorp/vault/api v1.0.4
go: extracting golang.org/x/sys v0.0.0-20200610111108-226ff32320da
go: extracting github.com/go-sql-driver/mysql v1.5.0
go: extracting github.com/hashicorp/vault/api v1.0.4
go: downloading github.com/minio/sha256-simd v0.1.1
go: downloading github.com/hashicorp/go-rootcerts v1.0.1
go: extracting github.com/hashicorp/go-rootcerts v1.0.1
go: downloading github.com/willf/bitset v1.1.10
go: extracting github.com/minio/sha256-simd v0.1.1
go: extracting github.com/willf/bitset v1.1.10
go: downloading github.com/modern-go/reflect2 v1.0.1
go: downloading github.com/hashicorp/go-cleanhttp v0.5.1
go: extracting github.com/hashicorp/go-cleanhttp v0.5.1
go: downloading github.com/streadway/amqp v0.0.0-20190404075320-75d898a42a94
go: downloading github.com/Shopify/sarama v1.24.1
go: extracting github.com/streadway/amqp v0.0.0-20190404075320-75d898a42a94
go: extracting golang.org/x/net v0.0.0-20200602114024-627f9648deb9
go: extracting github.com/modern-go/reflect2 v1.0.1
go: downloading github.com/hashicorp/vault/sdk v0.1.13
go: extracting github.com/hashicorp/vault/sdk v0.1.13
go: downloading go.uber.org/zap v1.14.1
go: downloading github.com/mitchellh/mapstructure v1.1.2
go: extracting github.com/mitchellh/mapstructure v1.1.2
go: downloading google.golang.org/genproto v0.0.0-20190819201941-24fa4b261c55
go: downloading github.com/spaolacci/murmur3 v1.1.0
go: extracting github.com/Shopify/sarama v1.24.1
go: extracting go.uber.org/zap v1.14.1
go: downloading github.com/eapache/go-resiliency v1.1.0
go: downloading github.com/klauspost/readahead v1.3.1
go: extracting github.com/spaolacci/murmur3 v1.1.0
go: downloading github.com/googleapis/gax-go/v2 v2.0.4
go: extracting google.golang.org/api v0.5.0
go: extracting github.com/eapache/go-resiliency v1.1.0
go: downloading github.com/nats-io/nats.go v1.10.0
go: extracting github.com/googleapis/gax-go/v2 v2.0.4
go: extracting github.com/klauspost/readahead v1.3.1
go: downloading github.com/eapache/queue v1.1.0
go: downloading github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: extracting github.com/nats-io/nats.go v1.10.0
go: downloading golang.org/x/text v0.3.2
go: extracting github.com/eapache/queue v1.1.0
go: downloading github.com/klauspost/reedsolomon v1.9.9
go: extracting github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: extracting github.com/klauspost/reedsolomon v1.9.9
go: downloading github.com/prometheus/client_golang v1.0.0
go: downloading github.com/hashicorp/go-retryablehttp v0.5.4
go: downloading github.com/hashicorp/go-multierror v1.0.0
go: extracting github.com/prometheus/client_golang v1.0.0
go: downloading github.com/prometheus/common v0.4.1
go: extracting github.com/hashicorp/go-multierror v1.0.0
go: extracting github.com/hashicorp/go-retryablehttp v0.5.4
go: downloading github.com/eclipse/paho.mqtt.golang v1.2.0
go: extracting github.com/prometheus/common v0.4.1
go: downloading github.com/minio/md5-simd v1.1.0
go: extracting github.com/eclipse/paho.mqtt.golang v1.2.0
go: extracting github.com/minio/md5-simd v1.1.0
go: downloading github.com/golang/snappy v0.0.1
go: extracting github.com/golang/snappy v0.0.1
go: downloading github.com/hashicorp/hcl v1.0.0
go: extracting google.golang.org/genproto v0.0.0-20190819201941-24fa4b261c55
go: extracting github.com/hashicorp/hcl v1.0.0
go: downloading github.com/nats-io/jwt v0.3.2
go: extracting github.com/nats-io/jwt v0.3.2
go: downloading go.uber.org/atomic v1.6.0
go: extracting go.uber.org/atomic v1.6.0
go: downloading github.com/lib/pq v1.7.0
go: extracting github.com/lib/pq v1.7.0
go: downloading github.com/pkg/errors v0.8.1
go: extracting github.com/pkg/errors v0.8.1
go: downloading github.com/golang/protobuf v1.4.0
go: extracting golang.org/x/text v0.3.2
go: extracting github.com/golang/protobuf v1.4.0
go: downloading github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: extracting github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: downloading gopkg.in/ldap.v3 v3.0.3
go: extracting gopkg.in/ldap.v3 v3.0.3
go: downloading github.com/rjeczalik/notify v0.9.2
go: extracting github.com/rjeczalik/notify v0.9.2
go: downloading github.com/hashicorp/errwrap v1.0.0
go: extracting github.com/hashicorp/errwrap v1.0.0
go: downloading github.com/pierrec/lz4 v2.4.0+incompatible
go: downloading github.com/minio/sio v0.2.0
go: extracting github.com/minio/sio v0.2.0
go: downloading github.com/rcrowley/go-metrics v0.0.0-20181016184325-3113b8401b8a
go: extracting github.com/rcrowley/go-metrics v0.0.0-20181016184325-3113b8401b8a
go: downloading gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: extracting gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: downloading github.com/matttproud/golang_protobuf_extensions v1.0.1
go: extracting github.com/matttproud/golang_protobuf_extensions v1.0.1
go: extracting gopkg.in/jcmturner/gokrb5.v7 v7.3.0
go: downloading github.com/beorn7/perks v1.0.0
go: downloading golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: downloading github.com/nats-io/nkeys v0.1.4
go: downloading github.com/gomodule/redigo v2.0.0+incompatible
go: downloading google.golang.org/protobuf v1.22.0
go: downloading github.com/klauspost/cpuid v1.3.0
go: extracting github.com/beorn7/perks v1.0.0
go: extracting github.com/nats-io/nkeys v0.1.4
go: downloading github.com/cespare/xxhash/v2 v2.1.1
go: downloading github.com/mattn/go-ieproxy v0.0.1
go: extracting github.com/cespare/xxhash/v2 v2.1.1
go: extracting github.com/mattn/go-ieproxy v0.0.1
go: downloading github.com/nsqio/go-nsq v1.0.7
go: extracting github.com/gomodule/redigo v2.0.0+incompatible
go: downloading github.com/prometheus/procfs v0.0.8
go: extracting golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: extracting github.com/prometheus/procfs v0.0.8
go: downloading github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: extracting github.com/klauspost/cpuid v1.3.0
go: extracting github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: downloading github.com/davecgh/go-spew v1.1.1
go: extracting github.com/nsqio/go-nsq v1.0.7
go: downloading gopkg.in/square/go-jose.v2 v2.3.1
go: extracting github.com/davecgh/go-spew v1.1.1
go: downloading github.com/fatih/structs v1.1.0
go: extracting github.com/fatih/structs v1.1.0
go: downloading gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: extracting gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: downloading gopkg.in/ini.v1 v1.42.0
go: extracting gopkg.in/square/go-jose.v2 v2.3.1
go: downloading gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: extracting gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: extracting gopkg.in/ini.v1 v1.42.0
go: downloading github.com/beevik/ntp v0.2.0
go: extracting google.golang.org/protobuf v1.22.0
go: extracting github.com/beevik/ntp v0.2.0
go: downloading github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: extracting github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: downloading go.opencensus.io v0.21.0
go: downloading github.com/coreos/go-semver v0.2.0
go: downloading gopkg.in/yaml.v2 v2.2.8
go: downloading go.uber.org/multierr v1.5.0
go: downloading github.com/prometheus/client_model v0.0.0-20190812154241-14fe0d1b01d4
go: extracting github.com/coreos/go-semver v0.2.0
go: downloading github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: extracting go.uber.org/multierr v1.5.0
go: downloading github.com/tinylib/msgp v1.1.2
go: extracting gopkg.in/yaml.v2 v2.2.8
go: downloading github.com/Azure/azure-storage-blob-go v0.8.0
go: extracting github.com/prometheus/client_model v0.0.0-20190812154241-14fe0d1b01d4
go: downloading github.com/dustin/go-humanize v1.0.0
go: extracting github.com/dustin/go-humanize v1.0.0
go: extracting go.opencensus.io v0.21.0
go: extracting github.com/tinylib/msgp v1.1.2
go: extracting github.com/Azure/azure-storage-blob-go v0.8.0
go: downloading github.com/rs/cors v1.7.0
go: downloading github.com/nats-io/nuid v1.0.1
go: extracting github.com/rs/cors v1.7.0
go: extracting github.com/nats-io/nuid v1.0.1
go: downloading github.com/minio/simdjson-go v0.1.5-0.20200303142138-b17fe061ea37
go: downloading git.apache.org/thrift.git v0.13.0
go: downloading golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: downloading github.com/gogo/protobuf v1.3.1
go: extracting golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: downloading github.com/hashicorp/golang-lru v0.5.1
go: extracting github.com/hashicorp/golang-lru v0.5.1
go: extracting github.com/gogo/protobuf v1.3.1
go: downloading github.com/alecthomas/participle v0.2.1
go: extracting github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: downloading gopkg.in/jcmturner/rpc.v1 v1.1.0
go: extracting gopkg.in/jcmturner/rpc.v1 v1.1.0
go: downloading github.com/fatih/color v1.7.0
go: extracting github.com/minio/simdjson-go v0.1.5-0.20200303142138-b17fe061ea37
go: extracting github.com/fatih/color v1.7.0
go: downloading github.com/mattn/go-runewidth v0.0.2
go: extracting github.com/mattn/go-runewidth v0.0.2
go: downloading github.com/klauspost/pgzip v1.2.1
go: extracting github.com/alecthomas/participle v0.2.1
go: downloading github.com/hashicorp/go-uuid v1.0.1
go: downloading github.com/tidwall/gjson v1.3.5
go: extracting github.com/hashicorp/go-uuid v1.0.1
go: downloading github.com/coredns/coredns v1.4.0
go: extracting github.com/tidwall/gjson v1.3.5
go: downloading github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: extracting github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: downloading github.com/philhofer/fwd v1.0.0
go: extracting github.com/philhofer/fwd v1.0.0
go: downloading github.com/coreos/go-systemd/v22 v22.0.0
go: extracting github.com/klauspost/pgzip v1.2.1
go: downloading github.com/ncw/directio v1.0.5
go: extracting github.com/ncw/directio v1.0.5
go: downloading github.com/nats-io/stan.go v0.7.0
go: extracting github.com/nats-io/stan.go v0.7.0
go: downloading github.com/tidwall/pretty v1.0.0
go: extracting github.com/tidwall/pretty v1.0.0
go: downloading github.com/mattn/go-colorable v0.1.4
go: extracting github.com/mattn/go-colorable v0.1.4
go: downloading github.com/tidwall/match v1.0.1
go: extracting github.com/coreos/go-systemd/v22 v22.0.0
go: extracting github.com/tidwall/match v1.0.1
go: downloading github.com/mattn/go-isatty v0.0.8
go: extracting github.com/mattn/go-isatty v0.0.8
go: downloading gopkg.in/jcmturner/aescts.v1 v1.0.1
go: downloading github.com/xdg/stringprep v1.0.0
go: extracting gopkg.in/jcmturner/aescts.v1 v1.0.1
go: downloading github.com/hashicorp/go-sockaddr v1.0.2
go: extracting github.com/xdg/stringprep v1.0.0
go: extracting github.com/coredns/coredns v1.4.0
go: downloading github.com/ryanuber/go-glob v1.0.0
go: extracting github.com/ryanuber/go-glob v1.0.0
go: extracting github.com/hashicorp/go-sockaddr v1.0.2
go: extracting git.apache.org/thrift.git v0.13.0
go: downloading github.com/tidwall/sjson v1.0.4
go: extracting github.com/tidwall/sjson v1.0.4
go: extracting github.com/pierrec/lz4 v2.4.0+incompatible
go: finding github.com/Azure/azure-pipeline-go v0.2.1
go: finding gopkg.in/jcmturner/gokrb5.v7 v7.3.0
go: finding github.com/colinmarc/hdfs/v2 v2.1.1
go: finding github.com/minio/sha256-simd v0.1.1
go: finding github.com/minio/minio-go/v6 v6.0.58-0.20200612001654-a57fec8037ec
go: finding github.com/minio/cli v1.22.0
go: finding github.com/montanaflynn/stats v0.5.0
go: finding github.com/dustin/go-humanize v1.0.0
go: finding github.com/Azure/azure-storage-blob-go v0.8.0
go: finding google.golang.org/api v0.5.0
go: finding github.com/minio/highwayhash v1.0.0
go: finding github.com/valyala/tcplisten v0.0.0-20161114210144-ceec8f93295a
go: finding golang.org/x/crypto v0.0.0-20200622213623-75b288015ac9
go: finding github.com/json-iterator/go v1.1.10
go: finding cloud.google.com/go v0.39.0
go: finding github.com/gorilla/mux v1.7.4
go: finding github.com/mattn/go-isatty v0.0.8
go: finding github.com/mattn/go-colorable v0.1.4
go: finding github.com/fatih/color v1.7.0
go: finding github.com/dgrijalva/jwt-go v3.2.0+incompatible
go: finding golang.org/x/sys v0.0.0-20200610111108-226ff32320da
go: finding github.com/mitchellh/go-homedir v1.1.0
go: finding github.com/golang/protobuf v1.4.0
go: finding gopkg.in/ini.v1 v1.42.0
go: finding github.com/hashicorp/go-uuid v1.0.1
go: finding github.com/rjeczalik/notify v0.9.2
go: finding github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd
go: finding github.com/jcmturner/gofork v0.0.0-20190328161633-dc7c13fece03
go: finding gopkg.in/jcmturner/dnsutils.v1 v1.0.1
go: finding github.com/modern-go/reflect2 v1.0.1
go: finding google.golang.org/protobuf v1.22.0
go: finding github.com/mattn/go-ieproxy v0.0.1
go: finding gopkg.in/jcmturner/goidentity.v3 v3.0.0
go: finding github.com/secure-io/sio-go v0.3.0
go: finding github.com/minio/md5-simd v1.1.0
go: finding golang.org/x/net v0.0.0-20200602114024-627f9648deb9
go: finding github.com/shirou/gopsutil v2.20.3-0.20200314133625-53cec6b37e6a+incompatible
go: finding gopkg.in/jcmturner/aescts.v1 v1.0.1
go: finding golang.org/x/oauth2 v0.0.0-20190226205417-e64efc72b421
go: finding github.com/ncw/directio v1.0.5
go: finding google.golang.org/grpc v1.26.0
go: finding golang.org/x/text v0.3.2
go: finding gopkg.in/jcmturner/rpc.v1 v1.1.0
go: finding google.golang.org/genproto v0.0.0-20190819201941-24fa4b261c55
go: finding go.opencensus.io v0.21.0
go: finding github.com/googleapis/gax-go/v2 v2.0.4
go: finding github.com/hashicorp/golang-lru v0.5.1
go: finding github.com/klauspost/cpuid v1.3.0
go: finding github.com/cespare/xxhash/v2 v2.1.1
go: finding github.com/cheggaaa/pb v1.0.28
go: finding github.com/mattn/go-runewidth v0.0.2
go: finding github.com/dchest/siphash v1.2.1
go: finding github.com/djherbis/atime v1.0.0
go: finding github.com/elazarl/go-bindata-assetfs v1.0.0
go: finding github.com/google/uuid v1.1.1
go: finding github.com/gorilla/handlers v1.4.2
go: finding github.com/gorilla/rpc v1.2.0
go: finding github.com/inconshreveable/go-update v0.0.0-20160112193335-8152e7eb6ccf
go: finding github.com/klauspost/readahead v1.3.1
go: finding github.com/klauspost/reedsolomon v1.9.9
go: finding github.com/miekg/dns v1.1.8
go: finding go.etcd.io/etcd/v3 v3.3.0-rc.0.0.20200707003333-58bb8ae09f8e
go: finding github.com/gogo/protobuf v1.3.1
go: finding go.uber.org/zap v1.14.1
go: finding go.uber.org/atomic v1.6.0
go: finding go.uber.org/multierr v1.5.0
go: finding github.com/coreos/go-systemd/v22 v22.0.0
go: finding github.com/coreos/go-semver v0.2.0
go: finding github.com/coredns/coredns v1.4.0
go: finding gopkg.in/ldap.v3 v3.0.3
go: finding gopkg.in/asn1-ber.v1 v1.0.0-20181015200546-f715ec2f112d
go: finding github.com/Shopify/sarama v1.24.1
go: finding github.com/davecgh/go-spew v1.1.1
go: finding github.com/eapache/go-resiliency v1.1.0
go: finding github.com/eapache/go-xerial-snappy v0.0.0-20180814174437-776d5712da21
go: finding github.com/golang/snappy v0.0.1
go: finding github.com/eapache/queue v1.1.0
go: finding github.com/pierrec/lz4 v2.4.0+incompatible
go: finding github.com/rcrowley/go-metrics v0.0.0-20181016184325-3113b8401b8a
go: finding github.com/eclipse/paho.mqtt.golang v1.2.0
go: finding github.com/go-sql-driver/mysql v1.5.0
go: finding github.com/gomodule/redigo v2.0.0+incompatible
go: finding github.com/lib/pq v1.7.0
go: finding github.com/nats-io/nats.go v1.10.0
go: finding github.com/nats-io/jwt v0.3.2
go: finding github.com/nats-io/nkeys v0.1.4
go: finding github.com/nats-io/nuid v1.0.1
go: finding github.com/nats-io/stan.go v0.7.0
go: finding github.com/nsqio/go-nsq v1.0.7
go: finding github.com/pkg/errors v0.8.1
go: finding github.com/streadway/amqp v0.0.0-20190404075320-75d898a42a94
go: finding github.com/xdg/scram v0.0.0-20180814205039-7eeb5667e42c
go: finding github.com/xdg/stringprep v1.0.0
go: finding gopkg.in/olivere/elastic.v5 v5.0.80
go: finding github.com/mailru/easyjson v0.0.0-20180730094502-03f2033d19d5
go: finding github.com/hashicorp/vault/api v1.0.4
go: finding github.com/hashicorp/errwrap v1.0.0
go: finding github.com/hashicorp/go-cleanhttp v0.5.1
go: finding github.com/hashicorp/go-multierror v1.0.0
go: finding github.com/hashicorp/go-retryablehttp v0.5.4
go: finding github.com/hashicorp/go-rootcerts v1.0.1
go: finding github.com/hashicorp/hcl v1.0.0
go: finding github.com/hashicorp/vault/sdk v0.1.13
go: finding github.com/hashicorp/go-sockaddr v1.0.2
go: finding github.com/ryanuber/go-glob v1.0.0
go: finding github.com/mitchellh/mapstructure v1.1.2
go: finding golang.org/x/time v0.0.0-20190308202827-9d24e82272b4
go: finding gopkg.in/square/go-jose.v2 v2.3.1
go: finding github.com/minio/sio v0.2.0
go: finding github.com/beevik/ntp v0.2.0
go: finding github.com/fatih/structs v1.1.0
go: finding gopkg.in/yaml.v2 v2.2.8
go: finding github.com/klauspost/pgzip v1.2.1
go: finding github.com/bcicen/jstream v0.0.0-20190220045926-16c1f8af81c2
go: finding github.com/alecthomas/participle v0.2.1
go: finding github.com/minio/simdjson-go v0.1.5-0.20200303142138-b17fe061ea37
go: finding git.apache.org/thrift.git v0.13.0
go: finding github.com/tidwall/gjson v1.3.5
go: finding github.com/tidwall/match v1.0.1
go: finding github.com/tidwall/pretty v1.0.0
go: finding github.com/tidwall/sjson v1.0.4
go: finding github.com/prometheus/client_golang v1.0.0
go: finding github.com/beorn7/perks v1.0.0
go: finding github.com/prometheus/client_model v0.0.0-20190812154241-14fe0d1b01d4
go: finding github.com/prometheus/common v0.4.1
go: finding github.com/matttproud/golang_protobuf_extensions v1.0.1
go: finding github.com/prometheus/procfs v0.0.8
go: finding github.com/rs/cors v1.7.0
go: finding github.com/tinylib/msgp v1.1.2
go: finding github.com/philhofer/fwd v1.0.0
go: finding github.com/willf/bloom v2.0.3+incompatible
go: finding github.com/spaolacci/murmur3 v1.1.0
go: finding github.com/willf/bitset v1.1.10
build github.com/minio/minio: cannot load github.com/klauspost/compress/flate: github.com/klauspost/compress@v1.10.3: reading https://proxy.golang.org/github.com/klauspost/compress/@v/v1.10.3.zip: 503 Service Unavailable
make: *** [build] Error 1

```

--- a/cmd/bucket-quota.go
+++ b/cmd/bucket-quota.go
@@ -196,8 +196,10 @@ func enforceFIFOQuota(ctx context.Context, objectAPI ObjectLayer) {
 		// Allocate new results channel to receive ObjectInfo.
 		objInfoCh := make(chan ObjectInfo)
 
+		versioned := globalBucketVersioningSys.Enabled(bucket)
+
 		// Walk through all objects
-		if err := objectAPI.Walk(ctx, bucket, "", objInfoCh); err != nil {
+		if err := objectAPI.Walk(ctx, bucket, "", objInfoCh, ObjectOptions{WalkVersions: versioned}); err != nil {
 			logger.LogIf(ctx, err)
 			continue
 		}
@@ -225,8 +227,6 @@ func enforceFIFOQuota(ctx context.Context, objectAPI ObjectLayer) {
 			}
 			scorer.addFileWithObjInfo(obj, 1)
 		}
-
-		versioned := globalBucketVersioningSys.Enabled(bucket)
 
 		var objects []ObjectToDelete
 		numKeys := len(scorer.fileObjInfos())

--- a/cmd/erasure-list-objects.go
+++ b/cmd/erasure-list-objects.go
@@ -53,6 +53,6 @@ func (er erasureObjects) HealObjects(ctx context.Context, bucket, prefix string,
 }
 
 // Walk - This is not implemented/needed anymore, look for erasure-zones.Walk()
-func (er erasureObjects) Walk(ctx context.Context, bucket, prefix string, results chan<- ObjectInfo) error {
+func (er erasureObjects) Walk(ctx context.Context, bucket, prefix string, results chan<- ObjectInfo, _ ObjectOptions) error {
 	return NotImplemented{}
 }

--- a/cmd/erasure-sets.go
+++ b/cmd/erasure-sets.go
@@ -855,8 +855,72 @@ func (f *FileInfoCh) Push(fi FileInfo) {
 	f.Valid = true
 }
 
-// Calculate least entry across multiple FileInfo channels,
-// returns the least common entry and the total number of times
+// Calculate lexically least entry across multiple FileInfo channels,
+// returns the lexically common entry and the total number of times
+// we found this entry. Additionally also returns a boolean
+// to indicate if the caller needs to call this function
+// again to list the next entry. It is callers responsibility
+// if the caller wishes to list N entries to call lexicallySortedEntry
+// N times until this boolean is 'false'.
+func lexicallySortedEntry(entryChs []FileInfoCh, entries []FileInfo, entriesValid []bool) (FileInfo, int, bool) {
+	for i := range entryChs {
+		entries[i], entriesValid[i] = entryChs[i].Pop()
+	}
+
+	var isTruncated = false
+	for _, valid := range entriesValid {
+		if !valid {
+			continue
+		}
+		isTruncated = true
+		break
+	}
+
+	var lentry FileInfo
+	var found bool
+	for i, valid := range entriesValid {
+		if !valid {
+			continue
+		}
+		if !found {
+			lentry = entries[i]
+			found = true
+			continue
+		}
+		if entries[i].Name < lentry.Name {
+			lentry = entries[i]
+		}
+	}
+
+	// We haven't been able to find any lexically least entry,
+	// this would mean that we don't have valid entry.
+	if !found {
+		return lentry, 0, isTruncated
+	}
+
+	lexicallySortedEntryCount := 0
+	for i, valid := range entriesValid {
+		if !valid {
+			continue
+		}
+
+		// Entries are duplicated across disks,
+		// we should simply skip such entries.
+		if lentry.Name == entries[i].Name && lentry.ModTime.Equal(entries[i].ModTime) {
+			lexicallySortedEntryCount++
+			continue
+		}
+
+		// Push all entries which are lexically higher
+		// and will be returned later in Pop()
+		entryChs[i].Push(entries[i])
+	}
+
+	return lentry, lexicallySortedEntryCount, isTruncated
+}
+
+// Calculate lexically least entry across multiple FileInfo channels,
+// returns the lexically common entry and the total number of times
 // we found this entry. Additionally also returns a boolean
 // to indicate if the caller needs to call this function
 // again to list the next entry. It is callers responsibility
@@ -892,7 +956,7 @@ func lexicallySortedEntryVersions(entryChs []FileInfoVersionsCh, entries []FileI
 		}
 	}
 
-	// We haven't been able to find any least entry,
+	// We haven't been able to find any lexically least entry,
 	// this would mean that we don't have valid entry.
 	if !found {
 		return lentry, 0, isTruncated
@@ -1508,32 +1572,58 @@ func (s *erasureSets) ListBucketsHeal(ctx context.Context) ([]BucketInfo, error)
 // to allocate a receive channel for ObjectInfo, upon any unhandled
 // error walker returns error. Optionally if context.Done() is received
 // then Walk() stops the walker.
-func (s *erasureSets) Walk(ctx context.Context, bucket, prefix string, results chan<- ObjectInfo) error {
+func (s *erasureSets) Walk(ctx context.Context, bucket, prefix string, results chan<- ObjectInfo, opts ObjectOptions) error {
 	if err := checkListObjsArgs(ctx, bucket, prefix, "", s); err != nil {
 		// Upon error close the channel.
 		close(results)
 		return err
 	}
 
-	entryChs := s.startMergeWalksVersions(ctx, bucket, prefix, "", true, ctx.Done())
+	if opts.WalkVersions {
+		entryChs := s.startMergeWalksVersions(ctx, bucket, prefix, "", true, ctx.Done())
+
+		entriesValid := make([]bool, len(entryChs))
+		entries := make([]FileInfoVersions, len(entryChs))
+
+		go func() {
+			defer close(results)
+
+			for {
+				entry, quorumCount, ok := lexicallySortedEntryVersions(entryChs, entries, entriesValid)
+				if !ok {
+					return
+				}
+
+				if quorumCount >= s.drivesPerSet/2 {
+					// Read quorum exists proceed
+					for _, version := range entry.Versions {
+						results <- version.ToObjectInfo(bucket, version.Name)
+					}
+				}
+				// skip entries which do not have quorum
+			}
+		}()
+
+		return nil
+	}
+
+	entryChs := s.startMergeWalks(ctx, bucket, prefix, "", true, ctx.Done())
 
 	entriesValid := make([]bool, len(entryChs))
-	entries := make([]FileInfoVersions, len(entryChs))
+	entries := make([]FileInfo, len(entryChs))
 
 	go func() {
 		defer close(results)
 
 		for {
-			entry, quorumCount, ok := lexicallySortedEntryVersions(entryChs, entries, entriesValid)
+			entry, quorumCount, ok := lexicallySortedEntry(entryChs, entries, entriesValid)
 			if !ok {
 				return
 			}
 
 			if quorumCount >= s.drivesPerSet/2 {
 				// Read quorum exists proceed
-				for _, version := range entry.Versions {
-					results <- version.ToObjectInfo(bucket, version.Name)
-				}
+				results <- entry.ToObjectInfo(bucket, entry.Name)
 			}
 			// skip entries which do not have quorum
 		}

--- a/cmd/erasure-zones.go
+++ b/cmd/erasure-zones.go
@@ -1799,16 +1799,58 @@ func (z *erasureZones) HealBucket(ctx context.Context, bucket string, dryRun, re
 // to allocate a receive channel for ObjectInfo, upon any unhandled
 // error walker returns error. Optionally if context.Done() is received
 // then Walk() stops the walker.
-func (z *erasureZones) Walk(ctx context.Context, bucket, prefix string, results chan<- ObjectInfo) error {
+func (z *erasureZones) Walk(ctx context.Context, bucket, prefix string, results chan<- ObjectInfo, opts ObjectOptions) error {
 	if err := checkListObjsArgs(ctx, bucket, prefix, "", z); err != nil {
 		// Upon error close the channel.
 		close(results)
 		return err
 	}
 
-	var zonesEntryChs [][]FileInfoVersionsCh
+	if opts.WalkVersions {
+		var zonesEntryChs [][]FileInfoVersionsCh
+		for _, zone := range z.zones {
+			zonesEntryChs = append(zonesEntryChs, zone.startMergeWalksVersions(ctx, bucket, prefix, "", true, ctx.Done()))
+		}
+
+		var zoneDrivesPerSet []int
+		for _, zone := range z.zones {
+			zoneDrivesPerSet = append(zoneDrivesPerSet, zone.drivesPerSet)
+		}
+
+		var zonesEntriesInfos [][]FileInfoVersions
+		var zonesEntriesValid [][]bool
+		for _, entryChs := range zonesEntryChs {
+			zonesEntriesInfos = append(zonesEntriesInfos, make([]FileInfoVersions, len(entryChs)))
+			zonesEntriesValid = append(zonesEntriesValid, make([]bool, len(entryChs)))
+		}
+
+		go func() {
+			defer close(results)
+
+			for {
+				entry, quorumCount, zoneIndex, ok := lexicallySortedEntryZoneVersions(zonesEntryChs, zonesEntriesInfos, zonesEntriesValid)
+				if !ok {
+					// We have reached EOF across all entryChs, break the loop.
+					return
+				}
+
+				if quorumCount >= zoneDrivesPerSet[zoneIndex]/2 {
+					// Read quorum exists proceed
+					for _, version := range entry.Versions {
+						results <- version.ToObjectInfo(bucket, version.Name)
+					}
+				}
+
+				// skip entries which do not have quorum
+			}
+		}()
+
+		return nil
+	}
+
+	var zonesEntryChs [][]FileInfoCh
 	for _, zone := range z.zones {
-		zonesEntryChs = append(zonesEntryChs, zone.startMergeWalksVersions(ctx, bucket, prefix, "", true, ctx.Done()))
+		zonesEntryChs = append(zonesEntryChs, zone.startMergeWalks(ctx, bucket, prefix, "", true, ctx.Done()))
 	}
 
 	var zoneDrivesPerSet []int
@@ -1816,10 +1858,10 @@ func (z *erasureZones) Walk(ctx context.Context, bucket, prefix string, results 
 		zoneDrivesPerSet = append(zoneDrivesPerSet, zone.drivesPerSet)
 	}
 
-	var zonesEntriesInfos [][]FileInfoVersions
+	var zonesEntriesInfos [][]FileInfo
 	var zonesEntriesValid [][]bool
 	for _, entryChs := range zonesEntryChs {
-		zonesEntriesInfos = append(zonesEntriesInfos, make([]FileInfoVersions, len(entryChs)))
+		zonesEntriesInfos = append(zonesEntriesInfos, make([]FileInfo, len(entryChs)))
 		zonesEntriesValid = append(zonesEntriesValid, make([]bool, len(entryChs)))
 	}
 
@@ -1827,7 +1869,7 @@ func (z *erasureZones) Walk(ctx context.Context, bucket, prefix string, results 
 		defer close(results)
 
 		for {
-			entry, quorumCount, zoneIndex, ok := lexicallySortedEntryZoneVersions(zonesEntryChs, zonesEntriesInfos, zonesEntriesValid)
+			entry, quorumCount, zoneIndex, ok := lexicallySortedEntryZone(zonesEntryChs, zonesEntriesInfos, zonesEntriesValid)
 			if !ok {
 				// We have reached EOF across all entryChs, break the loop.
 				return
@@ -1835,11 +1877,8 @@ func (z *erasureZones) Walk(ctx context.Context, bucket, prefix string, results 
 
 			if quorumCount >= zoneDrivesPerSet[zoneIndex]/2 {
 				// Read quorum exists proceed
-				for _, version := range entry.Versions {
-					results <- version.ToObjectInfo(bucket, version.Name)
-				}
+				results <- entry.ToObjectInfo(bucket, entry.Name)
 			}
-
 			// skip entries which do not have quorum
 		}
 	}()

--- a/cmd/fs-v1.go
+++ b/cmd/fs-v1.go
@@ -1484,7 +1484,7 @@ func (fs *FSObjects) HealBucket(ctx context.Context, bucket string, dryRun, remo
 // to allocate a receive channel for ObjectInfo, upon any unhandled
 // error walker returns error. Optionally if context.Done() is received
 // then Walk() stops the walker.
-func (fs *FSObjects) Walk(ctx context.Context, bucket, prefix string, results chan<- ObjectInfo) error {
+func (fs *FSObjects) Walk(ctx context.Context, bucket, prefix string, results chan<- ObjectInfo, opts ObjectOptions) error {
 	return fsWalk(ctx, fs, bucket, prefix, fs.listDirFactory(), results, fs.getObjectInfo, fs.getObjectInfo)
 }
 

--- a/cmd/gateway-main.go
+++ b/cmd/gateway-main.go
@@ -57,7 +57,7 @@ func (l *GatewayLocker) NewNSLock(ctx context.Context, bucket string, objects ..
 }
 
 // Walk - implements common gateway level Walker, to walk on all objects recursively at a prefix
-func (l *GatewayLocker) Walk(ctx context.Context, bucket, prefix string, results chan<- ObjectInfo) error {
+func (l *GatewayLocker) Walk(ctx context.Context, bucket, prefix string, results chan<- ObjectInfo, opts ObjectOptions) error {
 	walk := func(ctx context.Context, bucket, prefix string, results chan<- ObjectInfo) error {
 		var marker string
 
@@ -85,7 +85,7 @@ func (l *GatewayLocker) Walk(ctx context.Context, bucket, prefix string, results
 		return nil
 	}
 
-	if err := l.ObjectLayer.Walk(ctx, bucket, prefix, results); err != nil {
+	if err := l.ObjectLayer.Walk(ctx, bucket, prefix, results, opts); err != nil {
 		if _, ok := err.(NotImplemented); ok {
 			return walk(ctx, bucket, prefix, results)
 		}

--- a/cmd/gateway-unsupported.go
+++ b/cmd/gateway-unsupported.go
@@ -186,7 +186,7 @@ func (a GatewayUnsupported) ListObjectsV2(ctx context.Context, bucket, prefix, c
 }
 
 // Walk - Not implemented stub
-func (a GatewayUnsupported) Walk(ctx context.Context, bucket, prefix string, results chan<- ObjectInfo) error {
+func (a GatewayUnsupported) Walk(ctx context.Context, bucket, prefix string, results chan<- ObjectInfo, opts ObjectOptions) error {
 	return NotImplemented{}
 }
 

--- a/cmd/object-api-interface.go
+++ b/cmd/object-api-interface.go
@@ -38,6 +38,7 @@ type GetObjectInfoFn func(ctx context.Context, bucket, object string, opts Objec
 type ObjectOptions struct {
 	ServerSideEncryption encrypt.ServerSide
 	Versioned            bool                    // indicates if the bucket is versioned
+	WalkVersions         bool                    // indicates if the we are interested in walking versions
 	VersionID            string                  // Specifies the versionID which needs to be overwritten or read
 	MTime                time.Time               // Is only set in POST/PUT operations
 	UserDefined          map[string]string       // only set in case of POST/PUT operations
@@ -80,7 +81,7 @@ type ObjectLayer interface {
 	ListObjectsV2(ctx context.Context, bucket, prefix, continuationToken, delimiter string, maxKeys int, fetchOwner bool, startAfter string) (result ListObjectsV2Info, err error)
 	ListObjectVersions(ctx context.Context, bucket, prefix, marker, versionMarker, delimiter string, maxKeys int) (result ListObjectVersionsInfo, err error)
 	// Walk lists all objects including versions, delete markers.
-	Walk(ctx context.Context, bucket, prefix string, results chan<- ObjectInfo) error
+	Walk(ctx context.Context, bucket, prefix string, results chan<- ObjectInfo, opts ObjectOptions) error
 
 	// Object operations.
 

--- a/cmd/web-handlers.go
+++ b/cmd/web-handlers.go
@@ -655,8 +655,9 @@ func (web *webAPIHandlers) RemoveObject(r *http.Request, args *RemoveObjectArgs,
 		return nil
 	}
 
-	versioned := globalBucketVersioningSys.Enabled(args.BucketName)
-
+	opts := ObjectOptions{
+		Versioned: globalBucketVersioningSys.Enabled(args.BucketName),
+	}
 	var err error
 next:
 	for _, objectName := range args.Objects {
@@ -690,7 +691,7 @@ next:
 				}
 			}
 
-			_, err = deleteObject(ctx, objectAPI, web.CacheAPI(), args.BucketName, objectName, r, ObjectOptions{})
+			_, err = deleteObject(ctx, objectAPI, web.CacheAPI(), args.BucketName, objectName, r, opts)
 			logger.LogIf(ctx, err)
 		}
 
@@ -723,7 +724,7 @@ next:
 		objInfoCh := make(chan ObjectInfo)
 
 		// Walk through all objects
-		if err = objectAPI.Walk(ctx, args.BucketName, objectName, objInfoCh); err != nil {
+		if err = objectAPI.Walk(ctx, args.BucketName, objectName, objInfoCh, ObjectOptions{}); err != nil {
 			break next
 		}
 
@@ -736,7 +737,6 @@ next:
 				}
 				objects = append(objects, ObjectToDelete{
 					ObjectName: obj.Name,
-					VersionID:  obj.VersionID,
 				})
 			}
 
@@ -746,7 +746,7 @@ next:
 			}
 
 			// Deletes a list of objects.
-			_, errs := deleteObjects(ctx, args.BucketName, objects, ObjectOptions{Versioned: versioned})
+			_, errs := deleteObjects(ctx, args.BucketName, objects, opts)
 			for _, err := range errs {
 				if err != nil {
 					logger.LogIf(ctx, err)
@@ -1030,6 +1030,7 @@ func (web *webAPIHandlers) Upload(w http.ResponseWriter, r *http.Request) {
 		writeWebErrorResponse(w, err)
 		return
 	}
+
 	if objectAPI.IsCompressionSupported() && isCompressible(r.Header, object) && size > 0 {
 		// Storing the compression metadata.
 		metadata[ReservedMetadataPrefix+"compression"] = compressionAlgorithmV2
@@ -1052,15 +1053,15 @@ func (web *webAPIHandlers) Upload(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 	}
+
 	pReader = NewPutObjReader(hashReader, nil, nil)
 	// get gateway encryption options
-	var opts ObjectOptions
-	opts, err = putOpts(ctx, r, bucket, object, metadata)
-
+	opts, err := putOpts(ctx, r, bucket, object, metadata)
 	if err != nil {
 		writeErrorResponseHeadersOnly(w, toAPIError(ctx, err))
 		return
 	}
+
 	if objectAPI.IsEncryptionSupported() {
 		if crypto.IsRequested(r.Header) && !HasSuffix(object, SlashSeparator) { // handle SSE requests
 			rawReader := hashReader
@@ -1545,7 +1546,7 @@ func (web *webAPIHandlers) DownloadZip(w http.ResponseWriter, r *http.Request) {
 		objInfoCh := make(chan ObjectInfo)
 
 		// Walk through all objects
-		if err := objectAPI.Walk(ctx, args.BucketName, pathJoin(args.Prefix, object), objInfoCh); err != nil {
+		if err := objectAPI.Walk(ctx, args.BucketName, pathJoin(args.Prefix, object), objInfoCh, ObjectOptions{}); err != nil {
 			logger.LogIf(ctx, err)
 			continue
 		}

